### ISSUE TITLE
[Symfony] Sonata Notification Bundle with enqueue spikes to 100% CPU randomly

### ISSUE THREAD

**Issue by KDederichs** [php-enqueue/enqueue-dev#357]
Hey,

I already made an issue with a lot of info over at the sonata notification repo already but I'm having an issue where the sonata:notification:start command will sometimes cause the process to be stuck and run at 100% CPU till it is killed manually.

I got a new relic trace of this here:

```
Custom	Enqueue\AmqpLib\AmqpConsumer::receiveBasicGet	230	989,000 ms	77%
Custom	Enqueue\AmqpLib\AmqpConsumer::receive	230	283,000 ms	22%
Custom	Sonata\NotificationBundle\Iterator\AMQPMessageIterator::next	230	6,350 ms	0%
Custom	PhpAmqpLib\Wire\IO\StreamIO::select	125	5,380 ms	0%
Custom	Sonata\NotificationBundle\Command\ConsumerHandlerCommand::execute	1	1,330 ms	0%
Custom	PhpAmqpLib\Channel\AbstractChannel::wait	132	181 ms	0%
Total time			1,280,000 ms	100%
```

There is a lot more info on the sonata issue, so I'm gonna link that here as well.

https://github.com/sonata-project/SonataNotificationBundle/issues/313

Also seems like every times it does that it is preceded by 

```
ErrorException: Noticed exception 'ErrorException' with message 'fwrite(): send of 24 bytes failed with errno=104 Connection reset by peer' in /var/app/current/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php:282

Sample stack trace
in PhpAmqpLib\Wire\IO\StreamIO::error_handler called at ? (?)
…var/app/current/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/
StreamIO.php (282)
…nt/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/
AbstractConnection.php (341)
…nt/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/
AbstractConnection.php (462)
…/current/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/
AbstractChannel.php (238)
…/app/current/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/
AMQPChannel.php (1003)
…:basic_get called at /var/app/current/vendor/enqueue/amqp-lib/
AmqpConsumer.php (144)
…eiveNoWait called at /var/app/current/vendor/enqueue/amqp-lib/
AmqpConsumer.php (180)
…veBasicGet called at /var/app/current/vendor/enqueue/amqp-lib/
AmqpConsumer.php (129)
…vendor/sonata-project/notification-bundle/src/Iterator/
AMQPMessageIterator.php (98)
…vendor/sonata-project/notification-bundle/src/Iterator/
AMQPMessageIterator.php (140)
…ndor/sonata-project/notification-bundle/src/Command/
ConsumerHandlerCommand.php (93)
…rrent/vendor/symfony/symfony/src/Symfony/Component/Console/Command/
Command.php (252)
…p/current/vendor/symfony/symfony/src/Symfony/Component/Console/
Application.php (956)
…dor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/
Application.php (83)
…p/current/vendor/symfony/symfony/src/Symfony/Component/Console/
Application.php (240)
…dor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/
Application.php (71)
…p/current/vendor/symfony/symfony/src/Symfony/Component/Console/
Application.php (148)
in Symfony\Component\Console\Application::run called at /var/app/current/bin/console (27)

```

**Comment by makasim** [comment]
It uses [receiveBasicGet method](https://github.com/php-enqueue/amqp-lib/blob/master/AmqpConsumer.php#L175) internally (same as the previous implementation) and it is not nice when it comes to CPU. It would be better to use `recieveBasicConsume` method instead. It does blocking call and wait for the message without consuming CPU at all. 


The method is configured by [recive_method option](https://github.com/php-enqueue/amqp-lib/blob/master/AmqpConnectionFactory.php#L49) on the transport level. Not sure there is a way to access it from SonataNotificationBundle atm.

**Comment by KDederichs** [comment]
Ok so I rewrote it to the enqueue symfony bundle with SQS to see if that works better.
Turns out it still has strange CPU issues (and doesn't auto close after the specified time).
![100 enqueue](https://user-images.githubusercontent.com/24696606/36095071-eda8115a-0ff0-11e8-8999-a78c16eb6f38.png)

Any idea what's preventing it from terminating after 2 minutes? (Also should I open a new issue for this?)
